build.gradle.kts

```diff
@@ -0,0 +1,187 @@
+    logger.info("Using custom build type: ${buildType}")
+}
+
+val cmakeConfigure = tasks.register("cmakeConfigure") {
+    var cmakeArgs = listOf(
+        "-B${buildDir}/cmake-build",
+        "-H${projectDir}",
+        "-DCMAKE_BUILD_TYPE=${buildType}",
+        "-DCMAKE_INSTALL_PREFIX=${buildDir}/cmake-build",
+        "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON",
+        "-DBUILD_DEPS=ON",
+        "-DBUILD_TESTING=OFF"
+    )
+
+    if (org.gradle.internal.os.OperatingSystem.current().isLinux()) {
+        libcryptoPath = "/opt/openssl"
+        // To set this, add -PlibcryptoPath=/path/to/openssl/home on the command line
+        if (project.hasProperty("libcryptoPath")) {
+            libcryptoPath = project.property("libcryptoPath").toString()
+            logger.info("Using project libcrypto path: ${libcryptoPath}")
+        }
+    }
+
+    if (libcryptoPath != null) {
+        cmakeArgs += listOf(
+            "-DLibCrypto_INCLUDE_DIR=${libcryptoPath}/include",
+            "-DLibCrypto_STATIC_LIBRARY=${libcryptoPath}/lib/libcrypto.a"
+        )
+    }
+
+    inputs.file("CMakeLists.txt")
+    outputs.file("${buildDir}/cmake-build/CMakeCache.txt")
+
+    doLast {
+        val argsStr = cmakeArgs.joinToString(separator=" ")
+        logger.info("cmake ${argsStr}")
+        exec {
+            executable("cmake")
+            args(cmakeArgs)
+        }
+    }
+}
+
+val cmakeBuild = tasks.register("cmakeBuild") {
+    dependsOn(cmakeConfigure)
+    inputs.file("CMakeLists.txt")
+    inputs.file("${buildDir}/cmake-build/CMakeCache.txt")
+    inputs.files(fileTree("src/native").matching {
+        include(listOf("**/*.c", "**/*.h"))
+    })
+    inputs.files(fileTree("aws-common-runtime").matching {
+        include(listOf("**/CMakeLists.txt", "**/*.c", "**/*.h"))
+    })
+    outputs.file("${buildDir}/cmake-build/lib/libaws-crt-jni.so")
+
+    var cmakeArgs = listOf(
+        "--build", "${buildDir}/cmake-build",
+        "--target", "all"
+    )
+
+    doLast {
+        val argsStr = cmakeArgs.joinToString(separator=" ")
+        logger.info("cmake ${argsStr}")
+        exec {
+            executable("cmake")
+            args(cmakeArgs)
+        }
+    }
+}
+
+sourceSets {
+    main {
+        java {
+            setSrcDirs(listOf("src/main/java"))
+        }
+        // include shared libraries built by cmake/CI/CD in the lib folder
+        resources {
+            srcDir("${buildDir}/cmake-build/lib")
+        }
+    }
+    test {
+        java {
+            setSrcDirs(listOf("src/test/java"))
+        }
+    }
+}
+
+java {
+    withJavadocJar()
+    withSourcesJar()
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+tasks.compileJava {
+    dependsOn(cmakeBuild)
+}
+
+tasks.test {
+    useJUnit()
+    testLogging {
+        events("passed", "skipped", "failed")
+    }
+    for (prop in listOf("certificate", "privatekey", "endpoint", "rootca")) {
```

Comment:
This is looking in settings.gradle for possible values?

Reply:
or ```-Pcertificate=mycert.pem``` on the command line